test/integration/Integration.Contracts/ItemCommandsTopic.cs

```diff
@@ -8,11 +8,11 @@ public ItemCommandsTopic()
         {
         }
 
-        public string TopicFullName => this.SetTopicName().TopicFullName;
+        public string TopicFullName => this.SetTopicName("ci").TopicFullName;
 
-        public TopicBuilder SetTopicName()
+        public TopicBuilder SetTopicName(string environment)
```

Comment:
I think this parameter should be passed in the constructor instead of here otherwise I'm not seeing how a user can use this class easily

Reply:
yeah good idea. I will change it.

Reply:
The easiest way is to have a concrete consumer do it like this:

public class ConcreteConsumer : Consumer
{
public ConcreteConsumer(config) : base(new ConcreteTopic(config.Environment)) {}
}

Reply:
We don't need to pass the topic config in the consumer ctor, if we make the TopicBuilder receive the TopicConfig as an argument. It would be like this.

```csharp
public class ItemCommandsTopic : TopicBase<ItemCommandsV1>
    {
        public override TopicBuilder SetupTopicBuilder()
        {
            return new TopicBuilder(new TopicConfig(nameof(CreateItemV1), MessageType.Commands, "test-app", "dev", 1));
        }
    }
```

the consumer would be

```csharp
public TestConsumer(IConsumerConfiguration configuration)
            : base(new ItemCommandsTopic(), configuration, NullLogger.Instance)
```

this way we configure the topic in its class.